src/python/pants/goal/run_tracker.py

```diff
@@ -390,3 +413,39 @@ def shutdown_worker_pool(self):
     N.B. This exists only for internal use and to afford for fork()-safe operation in pantsd.
     """
     SubprocPool.shutdown(self._aborted)
+
+  def report_target_info(self, scope, target, keys, val):
+    """Add target information to run_info under target_data.
+
+    Will Recursively construct a nested dict with the keys provided.
+
+    :param string scope: The scope for which we are reporting the information.
+    :param string target: The target for which we want to store information.
+    :param list of string keys: The keys that will be recursively
+           nested and pointing to the information being stored.
+    :param dict or string val: The value of the information being stored.
+
+    :API: public
+    """
+    def create_dict_with_nested_keys_and_val(value, index):
+      """ Recursively constructs a nested dictionary with the keys pointing to the value.
+
+      :param dict or string value: The value of the information being stored.
+      :param int index: The index into the list of keys.
+      :return: dict of nested keys leading to the value.
+      """
+      if index > 0:
+        new_val = {keys[index]: value}
+        create_dict_with_nested_keys_and_val(new_val, index - 1)
+      else:
+        return {keys[index]: value}
+
+    val_to_store = create_dict_with_nested_keys_and_val(val, len(keys) - 1)
+    target_data = self._target_data.get(target, None)
+    if target_data is None:
+      self._target_data.update({target: {scope: val_to_store}})
```

Comment:
This will stomp on any previous data for the given target, including any previous data for this scope and for any other scopes.  I assume this isn't what we want?